### DISCUSSION TITLE
Can you add binaries to releases with GitHub Actions?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#27009] [Actions]
<p>I currently search for a way to automatically add generated binaries (in my case Jar files) to a release, which was already created, meaning it doesn’t create a release and adds files, but instead listens for when I create a release and attaches the files to it.</p>
<p>Is there a way to achieve this? I found some workflows, but they all don’t really explain what parts are important and how you could get files from different tasks (in my case a task in build.gradle that builds the jars and then uploads it to bintray)</p>

**Comment by wipe2238** [comment]
<p>Something like this? <a href="https://github.com/actions/upload-release-asset" rel="nofollow noopener">https://github.com/actions/upload-release-asset</a></p>

**Comment by Andre601** [comment]
<p>Yes, but from what I understand would this require the release to be created within the same workflow, while in my case I want to add the files/assets to the release I myself create.</p>
<p>Like already mentioned do I already have a setup to create the binaries to then upload them to an external site and are now wondering if I could use those generated files to attach them to the release I created.</p>

**Comment by crazy-max** [comment]
<p>Check out <a href="https://github.com/softprops/action-gh-release" rel="nofollow noopener">https://github.com/softprops/action-gh-release</a></p>

**Comment by Andre601** [comment]
<p>I think you don’t get my point.</p>
<p>I want to add an asset to an already existing release (when I created it) and not create one from a push, which is what your answer and the answer before you suggest.</p>
<p>I already found a possible solution and look into it further.</p>

**Comment by Blacksmoke16** [comment]
<p><a class="mention" href="/andre601">@andre601</a> What did you come up with?  I was looking to do something similar; define a workflow that gets executed when a release is created, which would build/upload the dist binary to it.</p>

**Comment by Andre601** [comment, marked as answer]
<p>I started to use <a href="https://github.com/AButler/upload-release-assets" rel="nofollow noopener">AButler/upload-release-assets</a> as it allows me to define multiple files to upload.</p>
<p>I set up a <a href="https://github.com/botblock/JavaBotBlockAPI/blob/master/.github/workflows/release_actions.yml#L29-L36" rel="nofollow noopener">workflow</a>, which creates the files, zips them, and then I upload both the zip and the generated files to the release which works perfectly fine.</p>
<p>You just need to make sure that the name of the file you create and the one you set in the workflow have the exact same spelling, or else it will be skipped without any errors nor messages.</p>

**Comment by Blacksmoke16** [comment]
<p>To close the loop on this, <a href="https://github.com/Blacksmoke16/oq/pull/47/files#diff-082c28d748ad2e3eecc5508d740d9417R9-R29" rel="nofollow noopener">this</a> is what I ended up with.</p>
<p>I learned that you can access the webhook payload within your action, so I simply listen for when a release was created, build the binary, and grab the <code>upload_url</code> as a way to upload it.  Quite simple and works perfectly for my use case.</p>

**Comment by Felixoid** [comment]
<p>Thank you very much, <a class="mention" href="/blacksmoke16">@blacksmoke16</a>, this one must be accepted as a proper solution</p>
